app/models/supermarket.rb

```diff
@@ -1,3 +1,10 @@
 class Supermarket < ApplicationRecord
   has_many :items
+
+  validates :name, presence: true
+  validates :location, presence: true
+
+  def supermarket_customers
+    customers = Customer.joins(items: :supermarket).where("supermarkets.name = '#{self.name}'").distinct.order(:name)
```

Comment:
You don't need to create a variable here (unless you were using it to see what your query was doing in pry). Nice job. 

Another thing that you could add to this model is an additional relationship:
```ruby
has_many :customers, through: :items
```
If you added this, then you could start your AR query directly on customers:
```ruby
def supermarket_customers
  customers.distinct
end
```